### PR TITLE
A-796 use decco as sontatype dependency

### PR DESCRIPTION
**Pull request by shaileshp0110**

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -29,34 +27,50 @@ object scalanet extends ScoverageModule {
...
+      ivy"io.iohk::decco-auto:1.0-SNAPSHOT"
     )
 
+    override def repositories = {
+      super.repositories ++ Seq(
+        MavenRepository("https://oss.sonatype.org/content/repositories/snapshots")
+      )
+    }
+
     override def moduleDeps: Seq[JavaModule] =
```
Comment (jtownson) [inline build.sc:59]: Please check if you can remove this.

File: build.sc [hunk 2]
```diff
@@ -29,34 +27,50 @@ object scalanet extends ScoverageModule {
...
   )
 
-  def moduleDeps: Seq[JavaModule] = Seq(decco.build.src.io.iohk.decco) ++ super.moduleDeps
+  override def repositories = {
+    super.repositories ++ Seq(
+      MavenRepository("https://oss.sonatype.org/content/repositories/snapshots")
+    )
+  }
 
   def scoverageVersion = "1.3.1"
```
Comment (jtownson) [inline build.sc]: Can be deleted.

**Review by jtownson — APPROVED** [review]
Fine to remove scoverage for now. Just a couple of redundant lines that can be deleted.

File: build.sc [hunk 3]
```diff
@@ -4,15 +4,13 @@ import scalalib._
...
 import $ivy.`com.lihaoyi::mill-contrib-buildinfo:0.4.1`
-import mill.contrib.scoverage.ScoverageModule
+import coursier.maven.MavenRepository
 
-import $file.decco.build
-
-object scalanet extends ScoverageModule {
+object scalanet extends ScalaModule {
 
   def scalaVersion = "2.12.7"
```
Comment (zejaco) [inline build.sc:11]: See couple of duplicates in both scalanet and test objects. Maybe it would be nice to have a common object here with these variables?

**Review by krcz — COMMENTED** [review]
@shaileshp0110 I have already implemented it here: https://github.com/input-output-hk/scalanet/pull/36. Can we drop this PR as it won't be useful when mine is merged?

**Comment by shaileshp0110** [comment]
> @shaileshp0110 I have already implemented it here: #36. Can we drop this PR as it won't be useful when mine is merged?

@krcz Tried using your branch its having issue using the decco from the snapshot https://oss.sonatype.org/content/repositories/snapshots

**Comment by krcz** [comment]
@shaileshp0110 Maybe it used in the past, bu now everything seems to work:
https://circleci.com/gh/input-output-hk/scalanet/371?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

What is the issue you mention?

**Comment by shaileshp0110** [comment]
> @shaileshp0110 Maybe it used in the past, bu now everything seems to work:
> https://circleci.com/gh/input-output-hk/scalanet/371?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
> 
> What is the issue you mention?

Coverage cannot use the snapshot decco, but you have changed it to Scala module which should work

**Comment by shaileshp0110** [comment]
> > @shaileshp0110 Maybe it used in the past, bu now everything seems to work:
> > https://circleci.com/gh/input-output-hk/scalanet/371?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
> > What is the issue you mention?
> 
> SCoverage cannot use the snapshot decco, but you have changed it to Scala module which should work, I had tried it 1 hour before and in the morning as well, But Jacek was blocked so I had to unblock him

**Comment by shaileshp0110** [comment]
https://github.com/input-output-hk/scalanet/pull/36 becuse its  done in this PR

**Pull request closed without merging**